tcms/testcases/tests/test_models.py

```diff
@@ -28,6 +29,43 @@ def test_create_testcase_with_cyrillic_works_issue_1770(self):
         case.refresh_from_db()
         self.assertTrue(case.summary.endswith("кирилица"))
 
+class TestCaseCalculateExpectedDuration(TestCase):
+    """Test TestCase.expected_duration"""
+
+    def test_duration_is_calculated_correctly(self):
+        case_random_duration = TestCaseFactory()
+        setup_duration = case_random_duration.setup_duration
+        testing_duration = case_random_duration.testing_duration
+        expected = setup_duration + testing_duration
+
+        case_random_duration.save()
+
+        case_random_duration.refresh_from_db()
+        self.assertEqual(case_random_duration.expected_duration, expected)
+
+    def test_empty_setup_duration_is_calculated_correctly(self):
+        case_empty_setup_duration = TestCaseFactory(setup_duration = None)
+        testing_duration = case_empty_setup_duration.testing_duration
+        case_empty_setup_duration.save()
+
+        case_empty_setup_duration.refresh_from_db()
+        self.assertEqual(case_empty_setup_duration.expected_duration, testing_duration)
+
+    def test_empty_testing_duration_is_calculated_correctly(self):
+        case_empty_testing_duration = TestCaseFactory(testing_duration = None)
+        setup_duration = case_empty_testing_duration.setup_duration
+        case_empty_testing_duration.save()
+
+        case_empty_testing_duration.refresh_from_db()
+        self.assertEqual(case_empty_testing_duration.expected_duration, setup_duration)
+
+    def test_empty_durations_are_calculated_correctly(self):
+        case_no_durations = TestCaseFactory(setup_duration = None, testing_duration = None)
+        case_no_durations.save()
+
+        case_no_durations.refresh_from_db()
+        self.assertEqual(case_no_durations.expected_duration, None)
+
```

Comment:
All of these scenarios are essentially the same with different input data. This can be implemented with a single test method and parameterized with hard-coded values.  See: `test_test_execution_statuses_can_be_deleted_but_one_must_remain` for example.